In [31]:
import seaborn as sns
import pandas as pd

pd.set_option('display.max_columns', 500)

In [6]:
data = pd.read_csv("data/processed_data.csv")
data.head()

sofifa_id                                         player_url  short_name  \
0     167575  https://sofifa.com/player/167575/anton-shunin/...   A. Shunin   
1     205211  https://sofifa.com/player/205211/milan-borjan/...   M. Borjan   
2     244700  https://sofifa.com/player/244700/yuriy-dyupin/...   Y. Dyupin   
3     199904  https://sofifa.com/player/199904/jesse-joronen...  J. Joronen   
4     240225  https://sofifa.com/player/240225/matvey-safono...  M. Safonov   

             long_name  overall  potential  value_eur  wage_eur         dob  \
0         Anton Shunin       76         76        NaN       NaN  1987-01-27   
1         Milan Borjan       76         76        NaN       NaN  1987-10-23   
2         Yuriy Dyupin       74         74        NaN       NaN  1988-03-17   
3  Jesse Pekka Joronen       72         72        NaN       NaN  1993-03-21   
4       Matvey Safonov       72         82        NaN       NaN  1999-02-25   

   weight_kg  ...  GK  LB LM  LW LWB  RB  RM  RW RWB ST  
0         88  ...   1   0  0   0   0   0   0   0   0  0  
1         84  ...   1   0  0   0   0   0   0   0   0  0  
2         84  ...   1   0  0   0   0   0   0   0   0  0  
3         89  ...   1   0  0   0   0   0   0   0   0  0  
4         82  ...   1   0  0   0   0   0   0   0   0  0  

[5 rows x 184 columns]

In [37]:
columns_to_evaluate = "ls,st,rs,lw,lf,cf,rf,rw,lam,cam,ram,lm,lcm,cm,rcm,rm,lwb,ldm,cdm,rdm,rwb,lb,lcb,cb,rcb,rb,gk".split(",")
for column_to_evaluate in columns_to_evaluate:
    evaluated_column = data[column_to_evaluate].apply(lambda cell: eval(cell))
    data[column_to_evaluate] = evaluated_column

In [49]:
nan_sum = data.isna().sum()
nan_sum[nan_sum > 0]

value_eur                74
wage_eur                 61
club_team_id             61
club_position            61
club_jersey_number       61
release_clause_eur     1176
pace                   2132
shooting               2132
passing                2132
dribbling              2132
defending              2132
physic                 2132
goalkeeping_speed     17107
dtype: int64

Analisando essa tabela com os valores nulos podemos perceber que alguns jogadores do dataset não possuem um time atual, isso explica
`wage_eur`, `club_team_id`, `club_position`, `club_jersey_number` terem a mesma quantidade de valores nulos (61),
porém `value_eur` deveria seguir esse mesmo padrão, assim podemos dizer que 13 (74 - 61) jogadores que estão ativos não possuem um valor,
portanto precisamos prever esses dados.

Também podemos perceber que temos 2132 jogadores não possuem `pace`, `shooting`, `passing`, `dribbling`, `defending`, `physic`. Vamos analisar esses jogadores

In [68]:
missing_main_attr_data = data[data.pace.isna()]

n_rows = missing_main_attr_data.shape[0]
n_gk = missing_main_attr_data.GK.sum()
print(f"porcentagem de goleiros {n_gk/n_rows * 100}%")
missing_main_attr_data

porcentagem de goleiros 100.0%


sofifa_id                                         player_url  \
0        167575  https://sofifa.com/player/167575/anton-shunin/...   
1        205211  https://sofifa.com/player/205211/milan-borjan/...   
2        244700  https://sofifa.com/player/244700/yuriy-dyupin/...   
3        199904  https://sofifa.com/player/199904/jesse-joronen...   
4        240225  https://sofifa.com/player/240225/matvey-safono...   
...         ...                                                ...   
2127     257618  https://sofifa.com/player/257618/armin-gremsl/...   
2128     262424  https://sofifa.com/player/262424/sorin-mogosan...   
2129     262420  https://sofifa.com/player/262420/robert-popa/2...   
2130     210022  https://sofifa.com/player/210022/philip-sprint...   
2131     244417  https://sofifa.com/player/244417/julian-krahl/...   

       short_name            long_name  overall  potential  value_eur  \
0       A. Shunin         Anton Shunin       76         76        NaN   
1       M. Borjan         Milan Borjan       76         76        NaN   
2       Y. Dyupin         Yuriy Dyupin       74         74        NaN   
3      J. Joronen  Jesse Pekka Joronen       72         72        NaN   
4      M. Safonov       Matvey Safonov       72         82        NaN   
...           ...                  ...      ...        ...        ...   
2127    A. Gremsl         Armin Gremsl       63         66   475000.0   
2128  S. Mogoșanu       Sorin Mogoșanu       58         60   170000.0   
2129      R. Popa          Robert Popa       56         75   325000.0   
2130    P. Sprint        Philip Sprint       62         64   350000.0   
2131     J. Krahl         Julian Krahl       62         76   850000.0   

      wage_eur         dob  weight_kg  height_cm  club_team_id club_position  \
0          NaN  1987-01-27         88        191           NaN           NaN   
1          NaN  1987-10-23         84        196           NaN           NaN   
2          NaN  1988-03-17         84        187           NaN           NaN   
3          NaN  1993-03-21         89        197           NaN           NaN   
4          NaN  1999-02-25         82        192           NaN           NaN   
...        ...         ...        ...        ...           ...           ...   
2127    2000.0  1994-08-13         85        190      115716.0           RES   
2128    1000.0  1992-11-22         77        180      115716.0            GK   
2129     500.0  2003-03-05         83        190      115716.0           SUB   
2130     700.0  1993-06-27         98        196      115820.0            GK   
2131     500.0  2000-01-22         90        194      115820.0           SUB   

      club_jersey_number preferred_foot  weak_foot  skill_moves  \
0                    NaN           Left          3            1   
1                    NaN          Right          2            1   
2                    NaN          Right          3            1   
3                    NaN          Right          3            1   
4                    NaN          Right          2            1   
...                  ...            ...        ...          ...   
2127                94.0           Left          2            1   
2128                 1.0          Right          1            1   
2129                31.0          Right          2            1   
2130                 1.0          Right          2            1   
2131                13.0          Right          3            1   

      international_reputation      work_rate         body_type  \
0                            1  Medium/Medium       Lean (185+)   
1                            1  Medium/Medium     Normal (185+)   
2                            1  Medium/Medium     Normal (185+)   
3                            1  Medium/Medium     Normal (185+)   
4                            1  Medium/Medium     Normal (185+)   
...                        ...            ...               ...   
2127                         1  Medium/Medium     Nor

Como vimos que os goleiros não tem os atributos listados acima, mas possuem um overall em cada posição, seria interessante prever esses atributos.

In [69]:
missing_gk_speed_data = data[data.goalkeeping_speed.isna()]
sum_of_missing_data = missing_gk_speed_data.shape[0] + missing_main_attr_data.shape[0]
sum_of_missing_data, data.shape[0]

(19239, 19239)

Assim podemos ver que todos os jogadores que não são goleiros não possuem `goalkeeping_speed`